improvement: S3C-3790 configurable MPU concurrency

jonathan-gramain opened the pull request:
Add a configuration value to tune MPU part replication concurrency,
instead of a hard-coded value.

Note that in 7.x branches, only the CRR-style replication is affected,
but the patch is also applied to Zenko logic for MPU.

jonathan-gramain committed:
improvement: S3C-3790 configurable MPU concurrency

Add a configuration value to tune MPU part replication concurrency,
instead of a hard-coded value.

Note that in 7.x branches, only the CRR-style replication is affected,
but the patch is also applied to Zenko logic for MPU.

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/backbeat/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable` with contents from `w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable origin/development/8.1
 $ git merge origin/w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3790-configurableMpuConcurrencyTunable` into
`development/7.4`
* [w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable](https://github.com/scality/backbeat/compare/development/7.9...w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable) will be merged into `development/7.9` (pull request #1316)
* [w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable) will be merged into `development/8.1` (pull request #1317)
* [w/8.2/bugfix/S3C-3790-configurableMpuConcurrencyTunable](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/bugfix/S3C-3790-configurableMpuConcurrencyTunable) will be merged into `development/8.2` (pull request #1318)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

jonathan-gramain committed:
improvement: S3C-3790 fix test config

- make queueProcessor section a requirement in joi description

- add it to config.json of functional tests

- add missing mpuPartsConcurrency param to crafted config in queue
  processor functional tests

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable` with contents from `w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable`
and `development/8.1`.

Please resolve the conflict on **the integration branch** (`w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable
 $ git pull  # or "git reset --hard origin/w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable"
 $ git merge origin/development/8.1
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable
```

bert-e commented:
# History mismatch



Merge commit #36aefe5320950035b0a3819a058abd8c54c9130c on the integration branch
`w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable` is merging a branch which is neither the current
branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` nor the development branch
`development/7.9`.

It is likely due to a rebase of the branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable` with contents from `w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable`
and `development/8.1`.

Please resolve the conflict on **the integration branch** (`w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable
 $ git pull  # or "git reset --hard origin/w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable"
 $ git merge origin/development/8.1
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

bert-e commented:
# History mismatch



Merge commit #1b85544d528b3667311d7336cb6b2554083a55e9 on the integration branch
`w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable` is merging a branch which is neither the current
branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` nor the development branch
`development/7.9`.

It is likely due to a rebase of the branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #1b85544d528b3667311d7336cb6b2554083a55e9 on the integration branch
`w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable` is merging a branch which is neither the current
branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #1b85544d528b3667311d7336cb6b2554083a55e9 on the integration branch
`w/8.2/bugfix/S3C-3790-configurableMpuConcurrencyTunable` is merging a branch which is neither the current
branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` nor the development branch
`development/8.2`.

It is likely due to a rebase of the branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3790-configurableMpuConcurrencyTunable` into
`development/7.4`
* [w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable](https://github.com/scality/backbeat/compare/development/7.9...w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable) will be merged into `development/7.9` (pull request #1316)
* [w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable) will be merged into `development/8.1` (pull request #1317)
* [w/8.2/bugfix/S3C-3790-configurableMpuConcurrencyTunable](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/bugfix/S3C-3790-configurableMpuConcurrencyTunable) will be merged into `development/8.2` (pull request #1319)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

bert-e commented:
# History mismatch



Merge commit #557d20f6e668221cb6a333077dcaacd8b34bb421 on the integration branch
`w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable` is merging a branch which is neither the current
branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` nor the development branch
`development/7.9`.

It is likely due to a rebase of the branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #557d20f6e668221cb6a333077dcaacd8b34bb421 on the integration branch
`w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable` is merging a branch which is neither the current
branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #557d20f6e668221cb6a333077dcaacd8b34bb421 on the integration branch
`w/8.2/bugfix/S3C-3790-configurableMpuConcurrencyTunable` is merging a branch which is neither the current
branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` nor the development branch
`development/8.2`.

It is likely due to a rebase of the branch `bugfix/S3C-3790-configurableMpuConcurrencyTunable` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3790-configurableMpuConcurrencyTunable` into
`development/7.4`
* [w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable](https://github.com/scality/backbeat/compare/development/7.9...w/7.9/bugfix/S3C-3790-configurableMpuConcurrencyTunable) will be merged into `development/7.9` (pull request #1316)
* [w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/S3C-3790-configurableMpuConcurrencyTunable) will be merged into `development/8.1` (pull request #1317)
* [w/8.2/bugfix/S3C-3790-configurableMpuConcurrencyTunable](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/bugfix/S3C-3790-configurableMpuConcurrencyTunable) will be merged into `development/8.2` (pull request #1320)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

jonathan-gramain commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.9`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.9`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`




Please check the status of the associated issue S3C-3790.

Goodbye jonathan-gramain.
